backend/src/main/java/ca/gov/dtsstn/cdcp/api/service/SubscriptionService.java

```diff
@@ -8,32 +9,53 @@
 import ca.gov.dtsstn.cdcp.api.service.domain.Subscription;
 import ca.gov.dtsstn.cdcp.api.service.domain.mapper.SubscriptionMapper;
 
 @Service
 public class SubscriptionService {
 
-	private final SubscriptionMapper subscriptionMapper = Mappers.getMapper(SubscriptionMapper.class);
-
 	private final SubscriptionRepository subscriptionRepository;
 
-	public SubscriptionService(SubscriptionRepository repository) {
-		Assert.notNull(repository, "subscriptionRepository is required; it must not be null");
-		this.subscriptionRepository = repository;
+	SubscriptionMapper subscriptionMapper = Mappers.getMapper(SubscriptionMapper.class);
+
+	private final UserRepository userRepository;
+
+	public SubscriptionService(
+				SubscriptionRepository subscriptionRepository, 
+				UserRepository userRepository) {
+		Assert.notNull(subscriptionRepository, "subscriptionRepository is required; it must not be null");
+		Assert.notNull(userRepository, "userRepository is required; it must not be null");
+		this.subscriptionRepository = subscriptionRepository;
+		this.userRepository = userRepository;
 	}
 
-	public Subscription create(Subscription subscription) {
+	public void create(Subscription subscription, String userId) {
 		Assert.notNull(subscription, "subscription is required; it must not be null");
-		Assert.isNull(subscription.getId(), "subscription.id must be null when creating new instance");
-
-		return subscriptionMapper.fromEntity(subscriptionRepository.save(subscriptionMapper.toEntity(subscription)));
+		Assert.notNull(userId, "userId is required; it must not be null");
+		final var userEntity = userRepository.findById(userId).get();
+		final var subscriptions = userEntity.getSubscriptions();
+		if (subscriptions.size() == 0) {
+			userEntity.setSubscriptions(Arrays.asList(subscriptionMapper.toEntity(subscription)));				
+		} else {
+			subscriptions.add(subscriptionMapper.toEntity(subscription));
+			userEntity.setSubscriptions(subscriptions);			
+		}
+		userRepository.save(userEntity);
 	}
 
-	public Subscription update(Subscription subscription) {
+	public void update(Subscription subscription, String userId, String subscriptionId) {
 		Assert.notNull(subscription, "subscription is required; it must not be null");
-		final var originalSubscription = subscriptionRepository.findById(subscription.getId()).orElseThrow();
-		return subscriptionMapper.fromEntity(subscriptionRepository.save(subscriptionMapper.update(subscription, originalSubscription)));
+		Assert.notNull(subscriptionId, "subscriptionId is required; it must not be null");
+		final var userEntity = userRepository.findById(userId).get();
+		final var oldSubscriptionEntity = subscriptionRepository.findById(subscriptionId).get();
+		final var newSubscriptionEntity = subscriptionMapper.update(subscription, oldSubscriptionEntity);
+		final var subscriptions = userEntity.getSubscriptions();
+		subscriptions.remove(oldSubscriptionEntity);
+		subscriptions.add(newSubscriptionEntity);
```

Comment:
We should be able to use `stream` and `map` on the existing subscriptions for a user rather than removing/adding here:
```java
final var subscriptions = userEntity.getSubscriptions().stream()
		.map(sub -> sub.getId().equals(subscription.getId()) ? newSubscriptionEntity : sub)
		.collect(Collectors.toSet());
```